addon/model.js

```diff
@@ -290,7 +290,15 @@ export default class MegamorphicModel extends Ember.Object {
   }
 
   get _modelName() {
-    return this._internalModel.modelName;
+    return this.__internalModel.modelName;
+  }
+
+  get _internalModel() {
+    return this.baseModel ? this.baseModel._internalModel : this.__internalModel;
+  }
+
+  set _internalModel(internalModel) {
+    this.__internalModel = internalModel;
```

Comment:
Are we sure `get _internalModel() {}` should prefer `baseModel._internalModel` when present instead of `__internalModel` ? otherwise it seems we risk trolling ourselves by setting an _internalModel that we don't access back.